core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -655,15 +655,19 @@ private[spark] object Utils extends Logging {
       // created the directories already, and that they are secured so that only the
       // user has access to them.
       getYarnLocalDirs(conf).split(",")
+    } else if (conf.getenv("SPARK_LOCAL_DIRS") != null) {
```

Comment:
I agree with @vanzin.

Here's the order of precedence that I think makes most sense

- `SPARK_LOCAL_DIRS` environment variable
- `spark.local.dirs` configuration
- `MESOS_DIRECTORY` environment variable
- `java.io.tmpdir`

This way, if people don't care about the local directory they fallback to Mesos dir while in Mesos and `/tmp` outside. But if people *do* care, they can configure it. Seems to be the best of both worlds.

Thoughts?